schema.md

```diff
@@ -0,0 +1,577 @@
+# WES - WRC RO-Crate
+
+This document aims to detail a conversion between the [GA4GH WES profile](https://ga4gh.github.io/workflow-execution-service-schemas/docs/) to the [Workflow Run Crate](https://www.researchobject.org/workflow-run-crate/profiles/workflow_run_crate/) (WRC) extension of [Ro-Crates](https://www.researchobject.org/ro-crate/)
+
+## WORKFLOW RUN CRATE RO-CRATE FIELDS
+
+For a Json object to conform with the Ro-Crate specification there are certain minimal requirements. Further still the WRC estension has it's own requirements. Considering that, the following definitions aim to differentiate the diffrent types of fields and how they should be considered in each implementation.
```

Comment:
I'm not sure I understand completely, does this mean the CrateGen tool won't be stand alone? And it'll need to be one part of a library that generates the RO-Crate fields?

Reply:
Let's say I would like to avoid for this tool to do things that require user intervention. But anything that isn't either, as you say, a constant, or can be fetched from TES or some other API call, likely needs to be provided by the user. And some of the stuff that you described that relates to RO-Crates _in general_ (i.e., not the WRROC-specific stuff) may fall within that category.

Sure, we could avoid a CLI with options for the user to fill in RO-Crate descriptions and the like by instead defining some dummy defaults so that we can create self-contained RO-Crates - with the expectation that users would change these later on. But that's a bit risky - it might lead to bad quality RO-Crates! Also, that's not the core business of this tool, so shouldn't be high priority.

The more likely (and safer) use case in the real world is that there's already an RO-Crate in the system, maybe containing project, author, and funding information. So for now I think we should just focus on the the use case where the client passes an _existing_ RO-Crate, and we put the WRROC-related entities inside and fill up from the generic stuff what absolutely needs filling up.

Reply:
okay, that makes sense, i guess I can isolate those fields so we can fill only those.

Reply:
Perhaps we can add both options, allow the user to insert the descriptive fields (or whatever other fields they would like to fill in as well), then provide the partially created RO-Crate to the CLI tool to fill in the remaining fields. Like this they can also fill in fields that can be generated then we can ask which fields they would like to overwrite or keep as is.

We can also add the option to generate the RO-Crate and then fill in the descriptive fields later. Personally, I feel this would provide a better user experience as the outstanding fields will pretty much already be underlined for them. Also I imagine this tool will not likely be used among people with no technical knowledge, as people without some technical knowledge they are more likely to stick to web based tools with interactive UIs. So we can probably expect the core user base to understand that they need to fill in the required fields after the CLI tool generates the RO-Crate.